easie: clarify deprovisioning (#1887)
clerk/clerk-docs: docs/authentication/enterprise-connections/overview.mdx

Clerk prevents users linked to deprovisioned accounts in the OpenID provider from accessing the app.

> Before creating a new session token for an EASIE user, Clerk verifies that the user has not been removed from the OpenID provider (e.g. [suspended](https://support.google.com/a/answer/33312?hl=en) or [deleted](https://support.google.com/a/answer/33314?hl=en) via Google Workspace, or [deleted](https://learn.microsoft.com/en-us/entra/fundamentals/how-to-create-delete-users#delete-a-user) via Microsoft Entra), with a delay of up to 10 minutes. When deprovisioning is detected, Clerk revokes that user's existing sessions and responds with 401 Unauthorized to any requests to retrieve a new session token.

Reviewer: ```suggestion
Before creating a new session token for an EASIE user, Clerk verifies whether the user has been deprovisioned from their OpenID provider (e.g., [suspended](https://support.google.com/a/answer/33312?hl=en) or [deleted](https://support.google.com/a/answer/33314?hl=en) in Google Workspace, or [deleted](https://learn.microsoft.com/en-us/entra/fundamentals/how-to-create-delete-users#delete-a-user) in Microsoft Entra). This verification process might involve a delay of up to 10 minutes. If deprovisioning is detected, Clerk revokes that user's existing sessions and responds to any requests for a new session token with a 401 Unauthorized error.
```

> Before creating a new session token for an EASIE user, Clerk verifies that the user has not been removed from the OpenID provider (e.g. [suspendeded](https://support.google.com/a/answer/33312?hl=en) or [deleted](https://support.google.com/a/answer/33314?hl=en) via Google Workspace, or [deleted](https://learn.microsoft.com/en-us/entra/fundamentals/how-to-create-delete-users#delete-a-user) via Microsoft Entra), with a delay of up to 10 minutes. When deprovisioning is detected, Clerk revokes that user's existing sessions, and responds with 401 Unauthorized to any requests to retrieve a new session token.

Reviewer 2: ```suggestion
Before creating a new session token for an EASIE user, Clerk verifies that the user has not been removed from the OpenID provider (e.g. [suspendeded](https://support.google.com/a/answer/33312?hl=en) or [deleted](https://support.google.com/a/answer/33314?hl=en) via Google Workspace, or [deleted](https://learn.microsoft.com/en-us/entra/fundamentals/how-to-create-delete-users#delete-a-user) via Microsoft Entra), with a delay of up to 10 minutes. When deprovisioning is detected, Clerk revokes that user's existing sessions, and responds with `401 Unauthorized` to any requests to retrieve a new session token.
```

> Before creating a new session token for an EASIE user, Clerk verifies that the user has not been removed from the OpenID provider (e.g. [suspendeded](https://support.google.com/a/answer/33312?hl=en) or [deleted](https://support.google.com/a/answer/33314?hl=en) via Google Workspace, or [deleted](https://learn.microsoft.com/en-us/entra/fundamentals/how-to-create-delete-users#delete-a-user) via Microsoft Entra), with a delay of up to 10 minutes. When deprovisioning is detected, Clerk revokes that user's existing sessions, and responds with 401 Unauthorized to any requests to retrieve a new session token.

Reviewer 3: ```suggestion
Before creating a new session token for an EASIE user, Clerk verifies that the user has not been removed from the OpenID provider (e.g. [suspended](https://support.google.com/a/answer/33312?hl=en) or [deleted](https://support.google.com/a/answer/33314?hl=en) via Google Workspace, or [deleted](https://learn.microsoft.com/en-us/entra/fundamentals/how-to-create-delete-users#delete-a-user) via Microsoft Entra), with a delay of up to 10 minutes. When deprovisioning is detected, Clerk revokes that user's existing sessions and responds with 401 Unauthorized to any requests to retrieve a new session token.
```

Reviewer 3: Aren't we able to detect disabled user accounts for Microsoft and deprovision? Did something change here?

> Before creating a new session token for an EASIE user, Clerk verifies whether the user has been deprovisioned from their OpenID provider (e.g., [suspended](https://support.google.com/a/answer/33312?hl=en) or [deleted](https://support.google.com/a/answer/33314?hl=en) in Google Workspace, or [deleted](https://learn.microsoft.com/en-us/entra/fundamentals/how-to-create-delete-users#delete-a-user) in Microsoft Entra). This verification process might involve a delay of up to 10 minutes. If deprovisioning is detected, Clerk revokes that user's existing sessions and responds to any requests for a new session token with a 401 Unauthorized error.

Reviewer 4: ```suggestion
Before creating a new session token for an EASIE user, Clerk verifies whether the user has been deprovisioned from their OpenID provider (e.g., [suspended](https://support.google.com/a/answer/33312?hl=en) or [deleted](https://support.google.com/a/answer/33314?hl=en) in Google Workspace, or [deleted](https://learn.microsoft.com/en-us/entra/fundamentals/how-to-create-delete-users#delete-a-user) in Microsoft Entra). This verification process might involve a delay of up to 10 minutes. If deprovisioning is detected, Clerk revokes that user's existing sessions and responds to any requests for a new session token with a `401 Unauthorized` error.
```
